beacon-chain/forkchoice/protoarray/types.go

```diff
@@ -16,35 +17,39 @@ type ForkChoice struct {
 
 // Store defines the fork choice store which includes block nodes and the last view of checkpoint information.
 type Store struct {
-	pruneThreshold uint64              // do not prune tree unless threshold is reached.
-	justifiedEpoch types.Epoch         // latest justified epoch in store.
-	finalizedEpoch types.Epoch         // latest finalized epoch in store.
-	finalizedRoot  [32]byte            // latest finalized root in store.
-	nodes          []*Node             // list of block nodes, each node is a representation of one block.
-	nodesIndices   map[[32]byte]uint64 // the root of block node and the nodes index in the list.
-	canonicalNodes map[[32]byte]bool   // the canonical block nodes.
-	nodesLock      sync.RWMutex
+	pruneThreshold             uint64                                  // do not prune tree unless threshold is reached.
+	justifiedEpoch             types.Epoch                             // latest justified epoch in store.
+	finalizedEpoch             types.Epoch                             // latest finalized epoch in store.
+	finalizedRoot              [fieldparams.RootLength]byte            // latest finalized root in store.
+	proposerBoostRoot          [fieldparams.RootLength]byte            // latest block root that was boosted after being received in a timely manner.
+	previousProposerBoostRoot  [fieldparams.RootLength]byte            // previous block root that was boosted after being received in a timely manner.
+	previousProposerBoostScore uint64                                  // previous proposer boosted root score.
```

Comment:
These fields are not used outside of `applyWeightChanges` and therefore they should be variables in that function's scope.

Reply:
but they are used in different, independent calls to applyWeightChanges, so they must be persisted somewhere. The alternative is they become package globals, but that is less appealing

Reply:
oh, I thought they were only used in the loop